api-error-response.go

```diff
@@ -83,6 +83,32 @@ const (
 	reportIssue = "Please report this issue at https://github.com/minio/minio-go/issues."
 )
 
+type APIErrorCode string
+
+var (
+	ErrCodeNoSuchBucket          = APIErrorCode("NoSuchBucket")
```

Comment:
The problem with typed approach is that this is non-exhaustive it requires us to constantly track the meaning. Current approach is XML holds the value and you look for what is needed as and when needed. 

Current approach is simpler a typed approach is that we are documenting all kinds of errors that can happen and will happen and their associated meaning.  Just doing it for a subset of errors is not useful. If we do it we do it for all S3 API errors we might receive.